src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/SourceLauncher.java

```diff
@@ -259,13 +259,16 @@ private Class<?> execute(MemoryContext context, String[] mainArgs)
             }
         } catch (IllegalAccessException e) {
             throw new Fault(Errors.CantAccessMainMethod(mainClassName));
-        } catch (InvocationTargetException e) {
+        } catch (InvocationTargetException exception) {
             // remove stack frames for source launcher
-            int invocationFrames = e.getStackTrace().length;
-            Throwable target = e.getCause();
-            StackTraceElement[] targetTrace = target.getStackTrace();
-            target.setStackTrace(Arrays.copyOfRange(targetTrace, 0, targetTrace.length - invocationFrames));
-            throw e;
+            Throwable cause = exception.getCause();
+            if (cause == null) throw exception;
+            StackTraceElement[] elements = cause.getStackTrace();
```

Comment:
Note that `Throwable​::getStackTrace()` is not `final`, so it can be made to return `null` by overriding.
```suggestion
            StackTraceElement[] elements = cause.getStackTrace();
            if (elements == null) throw exception;
```